packages/hebao_v1/contracts/base/BaseWallet.sol

```diff
@@ -159,7 +159,10 @@ contract BaseWallet is ReentrancyGuard, AddressSet, Wallet
     function bindMethod(bytes4 _method, address _module)
         external
         override
-        nonReentrant
+        reentrantWhitelist(
+            this.bindMethod.selector,
+            this.setup.selector | this.addModule.selector | this.removeModule.selector
+        )
```

Comment:
Because the function is `onlyModule`, isn't this just basically trying to ensure the module works correctly? Does is really matter that the reentrancy happens through `setup`, `addModule` or `removeModule` at some point? It seems to me that `bindMethod` needs to be safe against reentrancy attacks regardless, how it got there isn't important.

Also, because it's `onlyModule` safety is actually guaranteed by the module, you could see `bindMethod` more as an internal method of a module, which doesn't really need guards against reentrancy because that's handled on a higher level at functions that are externally accessible.